src/devices/src/virtio/persist.rs

```diff
@@ -76,6 +88,7 @@ impl Persist for Queue {
 /// State of a VirtioDevice.
 #[derive(Debug, PartialEq, Versionize)]
 pub struct VirtioDeviceState {
+    pub device_type: u32,
```

Comment:
Why do we need `device_type` in the state ?

Reply:
`device_type` is not used in our (de)serialization but is part of the generic virtio specification.

We should save all state irrespective of the _current_ actual implementation which may change.

Reply:
I agree with that, I am questioning this because I think this cannot change in the future. The specific device ids are not going to changed.

Reply:
I'm thinking of allowing using it as a validator/identifier if we ever need it. This ties the current `VirtioDeviceState` to a specific virtio device type.

I don't have a clear scenario where this would be imperatively necessary, but I think it's better to be safe than sorry. It's just one `u32` per device.